### ISSUE TITLE
Result of getTypeOfLocation for `PropertyAccessExpression` and `ElementAccessExpression` may be different between 6 and 7

### ISSUE THREAD

**Issue by jet2jet** [microsoft/typescript-go#3938]
(I know the APIs are marked as unstable, but I'll create this issue since I don't know whether this is expected or unexpected)

## Steps to reproduce

https://github.com/jet2jet/tsgo-test

The test module is:

```ts
interface A {
  a: number;
  b: { c: number };
}
const obj: A = { a: 1, b: { c: 2 } };
const a1 = obj.a;
// prettier-ignore
const a2 = (obj).a;
const c = obj.b.c;
```

The test code uses `getTypeAtLocation` for expressions `obj.a`, `(obj).a`, and `obj.b.c`.

## Behavior with `typescript@6.0`

All types of `obj.a`, `(obj).a`, and `obj.b.c` are reported as `number`.

```
---a1---
expr: PropertyAccessExpression obj.a
type: number
---a2---
expr: PropertyAccessExpression (obj).a
type: number
---c---
expr: PropertyAccessExpression obj.b.c
type: number
```

## Behavior with `tsgo`

Only a type of `obj.a` is reported as `number`.

```
---a1---
expr: PropertyAccessExpression obj.a
type: number
---a2---
expr: PropertyAccessExpression (obj).a
type: A
---c---
expr: PropertyAccessExpression obj.b.c
type: { c: number; }
```

**Comment by jet2jet** [comment]
I found that `resolveNodeHandle` in session.go may return different node instance than caller-expected one, because the actual node position, start and end, is not checked (`ast.GetNodeAtPosition` only accepts start position):
https://github.com/microsoft/typescript-go/blob/1b81e8f62fca85a104c091068d978846d3cad182/internal/api/session.go#L1817-L1858

For more, I encountered that `getResolvedSignature` will crash if `CallExpression` with `PropertyAccessExpression` is passed as an parameter.  
`getResolvedSignature` uses `resolveNodeHandle`, and if `CallExpression` 's `expression` is something like `Expression`, `resolveNodeHandle` returns that expression instead of `CallExpression`, which violates [`checker.resolveSignature` 's expectation](https://github.com/microsoft/typescript-go/blob/52d3d0ff095128a84c07d53d0cd85a8c67db3820/internal/checker/checker.go#L8420).
<details>
<summary>The output stack trace is:</summary>

```
Error: panic: Unhandled case in resolveSignature
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:26 +0x5e
github.com/microsoft/typescript-go/internal/api.(*SyncConn).handleRequest.func1()
        github.com/microsoft/typescript-go/internal/api/conn_sync.go:75 +0x58
panic({0xd53660?, 0x117b2a0?})
        runtime/panic.go:860 +0x13a
github.com/microsoft/typescript-go/internal/checker.(*Checker).resolveSignature(0x20e9049c8db0?, 0x1192780?, 0x20e9047ce8f0?, 0x11875d8?)
        github.com/microsoft/typescript-go/internal/checker/checker.go:8435 +0xb5
github.com/microsoft/typescript-go/internal/checker.(*Checker).getResolvedSignature(0x20e9049c8608, 0x20e9047ce8f0, 0x0, 0x0)
        github.com/microsoft/typescript-go/internal/checker/checker.go:8395 +0xd8
github.com/microsoft/typescript-go/internal/checker.(*Checker).GetResolvedSignature(...)
        github.com/microsoft/typescript-go/internal/checker/exports.go:221
github.com/microsoft/typescript-go/internal/api.(*Session).handleGetResolvedSignature(0x20e903ffe0a0, {0x11875d8?, 0x20e9040a4b00?}, 0x20e906566ba0)
        github.com/microsoft/typescript-go/internal/api/session.go:1030 +0xc5
github.com/microsoft/typescript-go/internal/api.(*Session).HandleRequest(0x20e903ffe0a0, {0x11875d8, 0x20e9040a4b00}, {0x20e90408cae0, 0x14}, {0x20e9065362d0?, 0xe0c8c0?, 0x20e903f7e001?})
        github.com/microsoft/typescript-go/internal/api/session.go:361 +0x1a05
github.com/microsoft/typescript-go/internal/api.(*SyncConn).handleRequest(0x20e903f64180, {0x11875d8?, 0x20e9040a4b00?}, 0x20e9042fb630)
        github.com/microsoft/typescript-go/internal/api/conn_sync.go:91 +0xb4
github.com/microsoft/typescript-go/internal/api.(*SyncConn).Run(0x20e903f64180, {0x11875d8, 0x20e9040a4b00})
        github.com/microsoft/typescript-go/internal/api/conn_sync.go:57 +0xff
github.com/microsoft/typescript-go/internal/api.(*StdioServer).Run(0x20e9040ade20, {0x11875d8, 0x20e9040a4b00})
        github.com/microsoft/typescript-go/internal/api/server.go:113 +0x68f
main.runAPI({0x20e9040020a0, 0x2, 0x2})
        github.com/microsoft/typescript-go/cmd/tsgo/api.go:54 +0x3a9
main.runMain()
        github.com/microsoft/typescript-go/cmd/tsgo/main.go:22 +0xa5
main.main()
        github.com/microsoft/typescript-go/cmd/tsgo/main.go:11 +0x13
```
</details>
